File: EMR_BMED.Backend/Utils/CsvUtils.cs
using System.Globalization;
using CsvHelper;
using CsvHelper.Configuration.Attributes;
using EMR_BMED.Backend.Models;

namespace EMR_BMED.Backend.Utils
{
  public static class CsvUtils
  {
    public static MedicationModel[] ImportMeds()
    {
      using StreamReader streamReader = new("meds.csv");
      using CsvReader csvReader = new(
        streamReader, CultureInfo.InvariantCulture);

      return csvReader
      .GetRecords<ImportedMed>()
      .Select((med) => new MedicationModel()
      {
        Name = med.Name,
        Form = med.Form,
        Method = "",
        IsPresRequired = med.IsPresRequired == "PR",
        Brand = med.Brand,
        Storing = med.Storing,
      })
      .ToArray();
    }
  }

  class ImportedMed
  {
    [Name("Denumire comerciala")]
    public required string Name { get; set; }

    [Name("Forma farmaceutica")]
    public required string Form { get; set; }

    [Name("Prescriptie")]
    public required string IsPresRequired { get; set; }

    [Name("Firma / tara producatoare APP")]
    public string? Brand { get; set; }

    [Name("Ambalaj")]
    public string? Storing { get; set; }
  }
}

File: EMR_BMED.Backend.JS/utils/csv.utils.js
import fs from "node:fs";
import * as csv from "csv-parse/sync";

const importMeds = () => {
  const stream = fs.readFileSync("./data/meds.csv");
  const meds = csv.parse(stream, {});

  const result = meds.slice(1).map((m) => {
    return {
      name: m[1],
      form: m[3],
      isPresRequired: m[9] === "PR",
      brand: m[5],
      storing: m[11],
    };
  });
  return result;
};

export default { importMeds };
